main/lsp/LSPPreprocessor.cc

```diff
@@ -186,13 +174,37 @@ unique_ptr<LSPMessage> LSPPreprocessor::makeAndCommitWorkspaceEdit(unique_ptr<So
     return newMsg;
 }
 
+bool LSPPreprocessor::ensureInitialized(LSPMethod method, const LSPMessage &msg) const {
```

Comment:
Logic moved from LSPLoop. Now, LSPPreprocessor is solely responsible for rejecting messages that are sent before initialization.